extensions/azurePublishNew/src/node/backgroundProcessManager.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { ProcessStatus } from '@botframework-composer/types';
+import { v4 as uuid } from 'uuid';
+
+export interface ProcessList {
+  [key: string]: ProcessStatus;
+}
+
+export class BackgroundProcessManager {
+  static processes: ProcessList = {};
+
+  static startProcess(
+    initialStatus: number,
+    projectId: string,
+    processName: string,
+    initialMessage?: string,
+    comment?: string
+  ): string {
+    const id = uuid();
+    this.processes[id] = {
+      id: id,
+      projectId: projectId,
+      processName: processName,
+      time: new Date(),
+      status: initialStatus,
+      message: initialMessage,
+      log: [initialMessage],
+      comment: comment,
+    };
+    return id;
+  }
+
+  static getStatus(id: string): ProcessStatus {
+    return this.processes[id];
+  }
+
+  /* Find a status by projectid/processname without knowing the id
+   * for example, find status of "default" publish target, which is localpublish for which there can only be 1 running process
+   * or find most recent publish to a given publish target
+   */
+  static getStatusByName(projectId: string, processName: string): ProcessStatus {
+    const matching = [];
```

Comment:
following can be replaced by filter and sort. Time is of type Date and can be sorted like b.time - a.time